sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -140,10 +143,16 @@ def on_request(self, request):
         :param request: The PipelineRequest object
         :type request: ~azure.core.pipeline.PipelineRequest
         """
-        request_id = request.context.options.pop('request_id', self._request_id)
-        if request_id:
-            header = {"x-ms-client-request-id":request_id}
-            request.http_request.headers.update(header)
+        if 'request_id' in request.context.options:
+            request_id = request.context.options.pop('request_id')
+        elif self._request_id:
+            request_id = self._request_id
+        elif self._auto_request_id:
+            request_id = str(uuid.uuid1())
+        else:
+            request_id = None
+        header = {"x-ms-client-request-id": request_id}
```

Comment:
Dont' set `None` if there is no request_id, but do nothng (keep the `if request_id` from the original code)

Reply:
In this case, user does not specify request_id and they set auto_request_id = False, in the spec, we don't send request id w/ the request. Right?

Reply:
Yes, if at the end "request_id" is None, don't add anything to headers. Be sure to NEVER add a None value header